src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.TranslationAction_Actions.cs

```diff
@@ -57,22 +58,22 @@ public override GeneratorDriver TransformGeneratorDriver(GeneratorDriver generat
                 {
                     // As we're merging ourselves with the prior touch action, we want to keep the old project state
                     // that we are translating from.
-                    return new TouchDocumentsAction(priorAction.OldProjectState, this.NewProjectState, _newStates);
+                    return new TouchDocumentsAction(priorAction.OldProjectState, NewProjectState, priorTouchAction._oldStates, _newStates);
                 }
 
                 return null;
             }
         }
 
-        internal sealed class TouchAdditionalDocumentAction(
+        internal sealed class TouchAdditionalDocumentsAction(
```

Comment:
Allow multiple documents to be changed at the same time, like TouchDocumentsAction.